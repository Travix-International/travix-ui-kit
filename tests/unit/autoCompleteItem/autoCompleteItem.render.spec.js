import * as enzyme from 'enzyme';
import * as enzymeToJson from 'enzyme-to-json';
import React from 'react';
import AutoCompleteItem from '../../../components/autoComplete/autoCompleteItem';

describe('AutoCompleteItem', () => {
  describe('#render()', () => {
    const shallow = enzyme.shallow;
    const shallowToJson = enzymeToJson.shallowToJson;

    it('should render correct autocomplete item', () => {
      const wrapper = shallow(
        <AutoCompleteItem
          value="value"
        >
          item
        </AutoCompleteItem>
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should not modify mods', () => {
      const mods = ['test'];
      const wrapper = shallow(
        <AutoCompleteItem
          mods={mods}
          value="value"
        >
          item
        </AutoCompleteItem>
      );
      expect(mods.length).toEqual(1);
      expect(mods[0]).toEqual('test');
      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should render autocomplete item with title mode', () => {
      const wrapper = shallow(
        <AutoCompleteItem
          isTitle
          value="value"
        >
          item
        </AutoCompleteItem>
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should render autocomplete item with active mode', () => {
      const wrapper = shallow(
        <AutoCompleteItem
          isActive
          value="value"
        >
          item
        </AutoCompleteItem>
      );

      expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
  });
});
