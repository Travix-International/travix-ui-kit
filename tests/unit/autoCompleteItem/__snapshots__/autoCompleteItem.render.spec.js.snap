// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoCompleteItem #render() should not modify mods 1`] = `
<li
  className="ui-autocomplete-item ui-autocomplete-item_test"
  mods={
    Array [
      "test",
    ]
  }
  onClick={[Function]}
  role="option"
  tabIndex="-1"
  value="value"
>
  item
</li>
`;

exports[`AutoCompleteItem #render() should render autocomplete item with active mode 1`] = `
<li
  className="ui-autocomplete-item ui-autocomplete-item_active"
  onClick={[Function]}
  role="option"
  tabIndex="-1"
  value="value"
>
  item
</li>
`;

exports[`AutoCompleteItem #render() should render autocomplete item with title mode 1`] = `
<li
  className="ui-autocomplete-item ui-autocomplete-item_title"
  onClick={false}
  role="option"
  tabIndex="-1"
  value="value"
>
  item
</li>
`;

exports[`AutoCompleteItem #render() should render correct autocomplete item 1`] = `
<li
  className="ui-autocomplete-item"
  onClick={[Function]}
  role="option"
  tabIndex="-1"
  value="value"
>
  item
</li>
`;
